tests/test_sagemaker.py

```diff
@@ -154,6 +154,79 @@ def test_delete_notebook_instance(self):
         self.assertTrue(notebook['NotebookInstanceStatus'], 'Deleting')
 
 
+class TestModelInstance(BaseTest):
+    def test_list_model(self):
+        session_factory = self.replay_flight_data(
+            'test_sagemaker_model')
+        p = self.load_policy({
+            'name': 'list-sagemaker-model',
+            'resource': 'sagemaker-model'
+        }, session_factory=session_factory)
+        resources = p.run()
+        self.assertGreaterEqual(len(resources), 1)
+
+    def test_delete_model(self):
+        session_factory = self.replay_flight_data(
+            'test_sagemaker_delete_model')
+        p = self.load_policy({
+            'name': 'delete-invalid-sagemaker-model',
+            'resource': 'sagemaker-model',
+            'filters': [
+                {'tag:DeleteMe': 'present'}],
+            'actions': [{'type': 'delete'}]}, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        p = self.load_policy({
+            'name': 'list-sagemaker-model',
+            'resource': 'sagemaker-model',
+            'filters': [
+                {'tag:DeleteMe': 'present'}]
+        }, session_factory=session_factory)
+        resources2 = p.run()
```

Comment:
fwiw generally we strongly prefer using the client api to validate a policies, actions rather than executing another policy (its typically significantly less record json output and more explicit), ie in this context just calling list tags for resource on model id.